### PR TITLE
Improve ci

### PR DESCRIPTION
**Pull request by Murmele**

### CODE REVIEW

**Comment by Murmele** [comment]
@kerizane I set the CI up. Do we need the windows binary? because otherwise a step can be removed and the CI will be much faster

Perhabs I am not able to test if the release creation works, because I don't have the permission.

**Comment by kerizane** [comment]
Yeah, I think we need the windows binary. How slow is it?

**Comment by kerizane** [comment]
So, there is a flatpak file and a flatpak yml. What does the yml do?

**Comment by kerizane** [comment]
Not sure what the best practice is here. What about if we only build the artifacts when we're doing a release? The only git actions that would run for PRs would be unittests and code quality. Although, I guess there's no harm in doing the builds and if someone manages to break one of them, it's better to know before we merge it...

I'd like to have a rolling unstable release, as well as the versioned releases. This could be done as a nightly (build run every 24 hours) or whenever we merge to development. 

What do you think?

**Comment by Murmele** [comment]
> Yeah, I think we need the windows binary. How slow is it?

It takes really long to upload. Why the windows installer is not enough?



> Not sure what the best practice is here. What about if we only build the artifacts when we're doing a release?

Having the artifcats has the benefit that also testers can easily install a new version. In Gittyup I am using this feature quite a lot. Just to test I am installing the flatpak version

**Comment by Murmele** [comment]
> So, there is a flatpak file and a flatpak yml. What does the yml do?

The yml is the manifest which was used to build the flatpak package. We can add it to the output files or not.

**Comment by Murmele** [comment]
> I'd like to have a rolling unstable release, as well as the versioned releases. This could be done as a nightly (build run every 24 hours) or whenever we merge to development.

Yes this is done here. Whenever there is a push to development or a tag is created on development, the publish job is executed.

  # a prerelase is created when pushing to master
    # a release is created when a tag will be set
    # last condition is the same as IS_RELEASE,
    # but environment variables cannot be used outside of steps
    # so it was copied to here too
    if: ${{ github.ref == 'refs/heads/development' || (github.event_name == 'push' && github.ref_type == 'tag' && startswith(github.ref_name, 'v'))  }}

It will look like this at the end:
https://github.com/Murmele/Gittyup/releases
image

**Comment by kerizane** [comment]
> It takes really long to upload. Why the windows installer is not enough?

I'm not too fussed, we can leave it out for now.

For reference, how long is "really long"? 10 mins, 1 hour?

**Comment by kerizane** [comment]
Would anyone want the yml file? What's the benefit in having it?

This looks great, thanks for doing this.

**Comment by Murmele** [comment]
> 

I think for now it is only a build artifact, but if someone wants to test something offline this file can be used.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        manifest-path: org.tuxemon.Tuxemon/org.tuxemon.Tuxemon.yaml
+        cache: false
+        branch: ${{ steps.flatpak_release_branch.outputs.value }}
+
+    - name: Publish build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        path: org.tuxemon.Tuxemon/*
+        name: Tuxemon Flatpak
+
```
Comment (Murmele) [inline .github/workflows/build.yml]:
```suggestion
```
Remove publishing the flatpak manifest in the artifacts

**Comment by Murmele** [comment]
> 
It takes about 14min to upload and 4 minutes to download again

**Review by kerizane — APPROVED** [review]

**Comment by Murmele** [comment]
Thanks for merging. Let's see if the publish works as intended :)

**Comment by kerizane** [comment]
It worked, amazing!

If you're interested there are a few more builds that need to be added, see https://github.com/Tuxemon/Tuxemon/issues/1126 and https://github.com/Tuxemon/Tuxemon/milestone/8

**Pull request merged**